src/shared/fdset.c

```diff
@@ -286,26 +330,27 @@ int fdset_close_others(FDSet *fds) {
 }
 
 unsigned fdset_size(FDSet *fds) {
-        return set_size(MAKE_SET(fds));
+        return hashmap_size(MAKE_HASHMAP(fds));
 }
 
 bool fdset_isempty(FDSet *fds) {
-        return set_isempty(MAKE_SET(fds));
+        return hashmap_isempty(MAKE_HASHMAP(fds));
 }
 
 int fdset_iterate(FDSet *s, Iterator *i) {
-        void *p;
+        const void *k;
+        void *v;
 
-        if (!set_iterate(MAKE_SET(s), i, &p))
+        if (!hashmap_iterate(MAKE_HASHMAP(s), i, &v, &k))
```

Comment:
this requires some tweaks, you need to skip the index entries, and go immediately to the next here, until you find the next fd